ff/serving/provider/offline.go

```diff
@@ -0,0 +1,116 @@
+package provider
+
+import (
+	"errors"
+	"time"
+)
+
+const (
+	MemoryOffline Type = "MEMORY_OFFLINE"
+)
+
+type OfflineResourceType int
+
+const (
+	None OfflineResourceType = iota
+	Label
+	Feature
+)
+
+type ResourceID struct {
+	Name, Variant string
+	Type          OfflineResourceType
+}
+
+func (id ResourceID) Check() error {
+	if id.Name == "" {
+		return errors.New("ResourceID must have Name set")
+	}
+	if id.Type != Label && id.Type != Feature {
+		return errors.New("Invalid ResourceID Type")
+	}
+	return nil
+}
+
+type OfflineStore interface {
+	CreateResourceTable(id ResourceID) (OfflineTable, error)
+	GetResourceTable(id ResourceID) (OfflineTable, error)
+}
+
+type ResourceRecord struct {
+	Entity string
+	Value  interface{}
+	TS     time.Time
+}
+
+func (rec ResourceRecord) Check() error {
+	if rec.Entity == "" {
+		return errors.New("ResourceRecord must have Entity set.")
```

Comment:
should it also have a timestamp?

Reply:
Also should every new table assert that the Value of of each ResourceRecord it contains be the same type?

Reply:
It should have a TS, good catch, the value asserts should come in with Schema which hasn't been implemented yet